### PR TITLE
Relative Link Updater

### PR DESCRIPTION
**Pull request by scbedd**
Updated to recursively discover as well.

@Azure/azure-sdk-eng

### CODE REVIEW

File: eng/pipelines/templates/scripts/replace-relative-links.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  BuildSHA: ''
+  RepoId: ''
+
+steps:
+  - task: PythonScript@0
+    displayName: Replace Relative Readme Links with Absolute References
+    inputs:
+      scriptSource: inline
+      script: |
+        import argparse
```
Comment (weshaggard) [inline eng/pipelines/templates/scripts/replace-relative-links.yml]: While I'm good with this for now I do worry about the size of this inline script and how we plan to maintain this vs the py script itself.

**Comment by weshaggard** [comment]
Did you also plan to add some logic to validate that all the links exist?

File: eng/pipelines/templates/scripts/replace-relative-links.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/scripts/replace-relative-links.yml]: You're right. I don't really have a good response right now :(

Comment (scbedd) [reply eng/pipelines/templates/scripts/replace-relative-links.yml]: Perhaps we put this out as a release, make the yaml _download_ that release, unzip, and call it?

**Comment by scbedd** [comment]
@weshaggard Yeah. I'll do that, and if the link doesn't resolve, leave the original. Good suggestion. I'll get it in over the weekend.

File: eng/pipelines/templates/scripts/replace-relative-links.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/scripts/replace-relative-links.yml]: Nothing to worry about right now but something to keep in the back of our mind to consider for clean-up especially if we end needing to make various changes to these scripts.

**Comment by weshaggard** [comment]
It would be nice to validate all the links not just the ones you are translating, but I'm not sure what you would do if there is a bad link. Perhaps it is worth failing in PR validation for those.

**Comment by scbedd** [comment]
It's a readme, so I'm actually in agreement that it should blow up.

**Pull request merged**